js/app/src/Blocks.svelte

```diff
@@ -518,6 +535,17 @@
 				messages = [new_message(data, -1, event), ...messages];
 			} else if (event == "clear_status") {
 				update_status(id, "complete", data);
+			} else if (event == "close_stream") {
```

Comment:
You'll see `close_stream` in two places in the front-end - 

1. The first is here where I've added a `close_stream` event that can be emitted by the svelte components. This signals to the queue that the stream has been closed client-side and so we can just terminate the event running on the queue
2. The second is the `close_stream` function in `init.ts`. If the stream is closed server-side (time limit reached), we need to signal to the components that they should stop streaming. Each streaming component exposes a `close_stream` callback that the app will call once the stream is closed server-side